pyelexon/datasets.py

```diff
@@ -1,7 +1,83 @@
-""" Functions for /datasets """
+import requests
+import pandas as pd
+from pyelexon.constants import BASE_URL
 
-""" 
- See https://developer.data.elexon.co.uk/api-details#api=prod-insol-insights-api for more information 
+class Datasets:
+
+    def fetch_data(self, endpoint, params=None, convert_to_dataframe=False):
+        """
+        Parameters:
+        - endpoint (str): The endpoint path to fetch data from.
+        - params (dict): Optional query parameters.
+        - convert_to_dataframe (bool): Flag to convert response to DataFrame (default: False).
+
+        Returns:
+        - dict or DataFrame: JSON response from the API or DataFrame if convert_to_dataframe is True.
+        """
+        url = f"{BASE_URL}{endpoint}"
+
+        try:
+            response = requests.get(url, params=params)
+            response.raise_for_status()  # Raise an exception for errors
+
+            if convert_to_dataframe:
+                # Check if response is a list of dictionaries
+                if isinstance(response.json(), list):
+                    df = pd.DataFrame(response.json())
+                else:
+                    df = pd.DataFrame(response.json()['data'])
+                return df
+            else:
+                return response.json()
+
+        except requests.exceptions.HTTPError as http_err:
+            print(f"HTTP error: {http_err}")
+        except Exception as err:
+            print(f"Other error: {err}")
+
+        return None
+
+    def fetch_abuc(self, publish_date_time_from, publish_date_time_to, convert_to_dataframe=True):
+        """
+        Parameters:
```

Comment:
sorry to be a pain, Just want to make good practice.

Could you add a link the api documentation for this route. 
This would allow the user to quickly click on the link and see more descirptions